tracer/contrib/elastictraced/elastictraced.go

```diff
@@ -0,0 +1,100 @@
+// Package elastictraced provides tracing for the Elastic Elasticsearch client.
+// Supports v3 (gopkg.in/olivere/elastic.v3), v5 (gopkg.in/olivere/elastic.v5)
+// but with v3 you must use `DoC` on all requests to capture the request context.
+package elastictraced
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// TracedTransport is a traced HTTP transport that captures Elasticsearch spans.
+type TracedTransport struct {
+	service string
+	tracer  *tracer.Tracer
+	*http.Transport
+}
+
+// RoundTrip satisfies the RoundTripper interface, wraps the sub Transport and
+// captures a span of the Elasticsearch request.
+func (t *TracedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if pathIsTraced(req.URL.Path) {
+		span := t.tracer.NewChildSpanFromContext("elasticsearch.query", req.Context())
+		span.Service = t.service
+		span.Type = ext.AppTypeDB
+		defer span.Finish()
+		span.SetMeta("elasticsearch.method", req.Method)
+		span.SetMeta("elasticsearch.url", req.URL.Path)
+		span.SetMeta("elasticsearch.params", req.URL.Query().Encode())
+
+		if req.Body != nil {
+			buf, err := ioutil.ReadAll(req.Body)
+			if err != nil {
+				return nil, err
+			}
+			span.SetMeta("elasticsearch.body", string(buf))
+			req.Body = ioutil.NopCloser(bytes.NewBuffer(buf))
+		}
+
+		// Run the request using the standard transport.
+		res, err := t.Transport.RoundTrip(req)
+
+		if err != nil {
+			span.SetError(err)
```

Comment:
Would `res.StatusCode` be available here? Would be nice to set it as a meta too.